frontend/src/metabase/dashboard/components/parameters/ParameterValueWidget.jsx

```diff
@@ -54,52 +57,88 @@ export default class ParameterValueWidget extends Component {
         }
     }
 
+    static getParameterIconName(parameterType) {
+        if (parameterType.search(/date/) !== -1) return "calendar";
+        if (parameterType.search(/location/) !== -1) return "location";
+        if (parameterType.search(/id/) !== -1) return "label";
+        return "clipboard";
+    }
+
+    constructor(props) {
```

Comment:
FYI I just discovered you can replace the whole constructor with `state = { isFocused: false }` if all you're doing is setting initial state.